### PR TITLE
[Do Not Merge][MNEDC] Client.config changed to YAML

### LINKED ISSUE #209: [Folder Structure] Follow standard Go project Layout
There is a popular [**standard** Go application project layout](https://github.com/golang-standards/project-layout) and many projects follow it.
I think we should change the folder structure in terms of this guide like below.
- `GoMain/src/main` > `cmd`
- `Buillder` > `build`
- `samples` > `examples`
- `samples/datastorage` > `configs/datastorage`
- `src` > `internal`
- `doc` > `docs`
- `doc/edge_orchestration_api.yaml` and `doc/edge_orchestration_api_secure.yaml` > `api/...`

Please add or rectify the list above since it might be incorrect!

### PR DESCRIPTION
**Pull request by sun-sharma**
Signed-off-by: Sunchit Sharma <sun.sharma@samsung.com>

# Description
client.config ->client-config.yaml , corresponding code changes have been made 

Fixes #209 

## Type of change

Please delete options that are not relevant.

- [X] Code cleanup/refactoring
- [X] Documentation update

# How Has This Been Tested?

1. Set Up one system in Main NAT as MNEDC server
2. Set up 2 system in Sub NAT as MNEDC Client
3. Checked the logs for connection
4. Requested Service from Sub NAT system to Main NAT system
5. Worked Properly as expected.

**Test Configuration**:

- Firmware version: Ubuntu 16.04
- Hardware: x86-64
- Edge Orchestration Release: Coconut

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: samples/mnedc/client-config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+server-ip: 192.168.0.125
```
Comment (t25kim) [inline samples/mnedc/client-config.yaml:1]: As @suresh-lc's [comment](https://github.com/lf-edge/edge-home-orchestration-go/issues/209#issuecomment-762714187), put this file in the `configs`.
Also, can you add some helpful comments on the first line like [LINK]?(https://github.com/envoyproxy/envoy/blob/main/configs/terminate_http1_connect.yaml)

File: src/controller/discoverymgr/mnedc/client/client.go [hunk 2]
```diff
@@ -442,19 +446,12 @@ func (c *Client) NotifyBroadcastServer(configPath string) error {
 		return err
 	}
 
-	file, err := os.Open(configPath)
+	serverIP, _, err := getMNEDCServerAddress(configPath)
 
 	if err != nil {
-		log.Println(logPrefix, "cant read config file from", configPath, err.Error())
+		log.Println(errors.New("Cannot read config file," + err.Error()))
```
Comment (t25kim) [inline src/controller/discoverymgr/mnedc/client/client.go:452]: Do you have any reason to use error.New() ?

File: src/controller/discoverymgr/mnedc/client/client.go [hunk 3]
```diff
@@ -98,6 +97,11 @@ type MNEDCClient interface {
 	SetClient(clientAPI restclient.Clienter)
 }
 
+type conf struct {
+	ServerIp string `yaml:"server-ip"`
```
Comment (t25kim) [inline src/controller/discoverymgr/mnedc/client/client.go:101]: How about 'ServerIP'?

**Review by t25kim — COMMENTED** [review]
Please take a look my comments.
Thank you!

File: samples/mnedc/client-config.yaml (hunk 1 above)
Comment (sun-sharma) [reply samples/mnedc/client-config.yaml:1]: I will create the configs and examples in the next PR and move it.

> Also, can you add some helpful comments on the first line like [LINK]?

Yes, that is a good idea. I will do that.

File: src/controller/discoverymgr/mnedc/client/client.go (hunk 2 above)
Comment (sun-sharma) [reply src/controller/discoverymgr/mnedc/client/client.go:452]: No, no specific reason. Just did that to keep the flow consistent wherever I was reading YAML files. If you feel old one was better, I would do that. Please suggest !

File: src/controller/discoverymgr/mnedc/client/client.go (hunk 3 above)
Comment (sun-sharma) [reply src/controller/discoverymgr/mnedc/client/client.go:101]: Sure, I would change it. 'ServerIP' looks better.

**Comment by tdrozdovsky** [comment]
@sun-sharma Please finalize this request! I don't understand its status: completed or in progress?

**Comment by MoonkiHong** [comment]
> @sun-sharma Please finalize this request! I don't understand its status: completed or in progress?

@tdrozdovsky It is still in progress to rectify the root cause investigation. Please stay tuned.

**Comment by MoonkiHong** [comment]
This PR will be re-evaluated again once after all the regarding MNEDC refactoring/bug fixes are completed.

**Review by MoonkiHong — COMMENTED** [review]
This PR will be re-evaluated again once after all the regarding MNEDC refactoring/bug fixes are completed.

**Comment by sun-sharma** [comment]
Closing this, as new PR re-opened for the same.

**Pull request closed without merging**